### PR TITLE
[MSP430] Removed code which deletes indicator characters for certain asm syntaxes

### LINKED ISSUE #3324: MSP430 disassembly, missing indicator for Immediate Value or Memory Location
* Operating System: Windows 10
* Cutter version: 2.3.4
* Obtained from: Downloaded release from Cutter website or GitHub 
* File format: RAW Binary

After I have disassembled a MSP430 binary, I do not see an "#" or "(0x1234)" or "@" to show it is a Immediate Value or Memory Location.  It just shows the plain hex numbers.
There are some parenthesis that appear.  But they show only when Memory offsets are used.

Image5

### PR DESCRIPTION
**Pull request by moste00**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This should fix [issue #3324](https://github.com/rizinorg/cutter/issues/3324) in Cutter's bug log. 

...

**Test plan**

I have searched in the git history and tracked the introduction of the deleted code to commit hash [c83d5f4664f222f318f8e4ae3910aeb9e5515902](https://github.com/rizinorg/rizin/commit/c83d5f4664f222f318f8e4ae3910aeb9e5515902), I have tried to find any PR or detailed message associated with this hash, explaining why the code was needed, but I found none, the commit was probably added sometimes in the project's pre-history (back when it was radare2). 

As far as I can see, the deleted code has no other effect or purpose except the undesirable effect of deleting indicator characters from the disassembly when the asm syntax isn't AT&T. Any tests breaking due to the removal of this code are wrong tests that should be removed as well.

**To verify**

Run `rz-asm -a msp430 -d 3f401000` on dev and then on this PR's branch. On dev, it produces `mov 0x0010, r15` as output, which is wrong, the correct output should include the character '#' before the constant, as it appears on this PR's branch.

### CODE REVIEW

File: test/db/asm/msp430
Comment (XVilka) [inline test/db/asm/msp430:1]: Could you please add an instruction test with the `%` indicator as well?

**Review by XVilka — CHANGES_REQUESTED** [review]
LGTM with a small request

File: test/db/asm/msp430
Comment (moste00) [reply test/db/asm/msp430:1]: Hmmm, I don't think there is a `%` indicator in MSP430 assembly. If you Ctrl-F "%" in the disassembler source code (`librz\arch\isa\msp430\msp430_disas.c`), you will find only its usages as a format control character to the printing function, it's never used as an actual output of the disassembler.

Comment (moste00) [reply test/db/asm/msp430:1]: The indicators are only "#", "&", "@", and "@Rn+". The disassembler also seems to use the "$" indicator to imply a relative jump, but that usage is actually not-documented elsewhere in MSP430 resources, a relative jump is implied by just a plain address, which implicitly means the (ADDR + PC) addressing mode according the MSP430 manual.

**Review by XVilka — APPROVED** [review]

**Pull request merged**